rootfs/scheduler/__init__.py

```diff
@@ -481,59 +476,62 @@ def run(self, name, image, entrypoint, command, **kwargs):
             l["image"] = image
             l["slugimage"] = settings.SLUGRUNNER_IMAGE
 
-        template = string.Template(POD).substitute(l)
+        template = json.loads(string.Template(POD).substitute(l))
+
         if command.startswith('-c '):
             args = command.split(' ', 1)
             args[1] = args[1][1:-1]
         else:
             args = [command[1:-1]]
 
-        js_template = json.loads(template)
-        containers = js_template['spec']['containers'][0]
+        containers = template['spec']['containers'][0]
         containers['command'] = [entrypoint]
         containers['args'] = args
 
         self._set_environment(containers, **kwargs)
 
-        url = self._api("/namespaces/{}/pods", appname)
-        resp = self.session.post(url, json=js_template)
-        if unhealthy(resp.status_code):
-            error(resp, 'create Pod in Namespace "{}"', appname)
+        url = self._api("/namespaces/{}/pods", namespace)
+        response = self.session.post(url, json=template)
+        if unhealthy(response.status_code):
+            error(response, 'create Pod in Namespace "{}"', namespace)
 
         data = ''
         duration = 30
         iteration = 1
-        while(iteration < duration):
+        while (iteration < duration):
             try:
-                response = self._get_pod(appname, name)
+                response = self._get_pod(namespace, name)
                 data = response.text
                 pod = response.json()
                 if pod['status']['phase'] == 'Succeeded':
-                    response = self._pod_log(appname, name)
+                    response = self._pod_log(namespace, name)
                     response.encoding = 'utf-8'  # defaults to "ISO-8859-1" otherwise...
                     log = response.text
-                    self._delete_pod(appname, name)
+                    self._delete_pod(namespace, name)
                     return 0, log
+
                 if pod['status']['phase'] == 'Running':
                     if iteration > 28:
-                        duration = duration + 1
-            except:
+                        duration += 1
+            except KubeException:
                 break
```

Comment:
Likely, @smothiki was using this as a flow mechanism tho I think. I just made the exception catching more explicit